client-linuxapp/src/serviceinfo.rs

```diff
@@ -74,22 +74,37 @@ fn create_user(user: &str) -> Result<()> {
     Ok(())
 }
 
+fn check_password_encryption(s: &str) -> bool {
```

Comment:
it is nice to add a comment to the original source if you have been inspired by some other source code for this function